File: app/src/main/java/com/govnokoder/velotracker/ui/main/PageHistory.java
package com.govnokoder.velotracker.ui.main;

import android.content.Intent;
import android.graphics.Canvas;
import android.graphics.Color;
import android.graphics.ColorFilter;
import android.graphics.drawable.Drawable;
import android.os.Bundle;
import android.text.style.DrawableMarginSpan;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.ListView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.govnokoder.velotracker.BL.Controller.TrainingController;
import com.govnokoder.velotracker.BL.Model.Training;
import com.govnokoder.velotracker.LookTraining;
import com.govnokoder.velotracker.R;

import java.util.ArrayList;
import java.util.List;

public class PageHistory extends Fragment {

    private int pageNumber;
    private ListView listView;

    public static PageHistory newInstance(int page) {
        PageHistory fragment = new PageHistory();
        Bundle args=new Bundle();
        args.putInt("num", page);
        fragment.setArguments(args);
        return fragment;
    }

    public PageHistory() { }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        pageNumber = getArguments() != null ? getArguments().getInt("num") : 1;
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View result = inflater.inflate(R.layout.history_page, container, false);
        listView = (ListView) result.findViewById(R.id.listView);

        listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                Intent intent = new Intent(getContext(), LookTraining.class);
                intent.putExtra("INDEX", position);
                startActivity(intent);
            }
        });
        return result;
    }

    @Override
    public void onStart() {
        super.onStart();
    }

    @Override
    public void onResume() {
        super.onResume();
        TrainingController trainingController = new TrainingController(getContext());
        List<Training> trainings = trainingController.LoadTrainingsData();
        List<String> listDate = new ArrayList<>();
        if(trainings != null) {
            for (Training training: trainings) {
                listDate.add(training.Date.toString());
            }
        }
        ArrayAdapter<String> adapter = new ArrayAdapter(getContext(),
                android.R.layout.simple_list_item_1, listDate){
            @NonNull
            @Override
            public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
                View view = super.getView(position, convertView, parent);
                TextView textView = (TextView)view.findViewById(android.R.id.text1);
                textView.setTextColor(Color.WHITE);
                textView.setBackgroundColor(Color.argb(255, 0, 0, 0));
                return view;
            }
        };
        listView.setAdapter(adapter);
    }
}

File: app/src/main/java/com/coursework/velotracker/ui/main/PageHistory.kt
package com.coursework.velotracker.ui.main

import android.content.Context
import android.content.Intent
import android.graphics.Color
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.*
import androidx.fragment.app.Fragment
import com.coursework.velotracker.AppConstants
import com.coursework.velotracker.BL.Controller.TrainingController
import com.coursework.velotracker.BL.Model.Extensions.toStringExtension
import com.coursework.velotracker.BL.Model.Training.TrainingStatistics
import com.coursework.velotracker.LookTraining
import com.coursework.velotracker.R
import com.coursework.velotracker.databinding.ActivityTrainingBinding
import com.coursework.velotracker.databinding.HistoryPageBinding
import kotlinx.coroutines.GlobalScope
import kotlinx.coroutines.launch
import java.util.*

class PageHistory: Fragment(), AdapterView.OnItemClickListener {

    private var _binding: HistoryPageBinding? = null
    private val binding get() = _binding!!
    private var pageNumber: Int = 1

    companion object{
        fun newInstance(page: Int):PageHistory {
            val fragment = PageHistory()
            val args = Bundle()
            args.putInt("num", page)
            fragment.arguments = args
            return fragment
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        pageNumber = arguments?.getInt("num")?:1
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        _binding = HistoryPageBinding.inflate(inflater, container, false)
        binding.listView.onItemClickListener = this
        return binding.root
    }

    override fun onItemClick(parent: AdapterView<*>?, view: View?, position: Int, id: Long) {
        val intent = Intent(context, LookTraining::class.java)
        intent.putExtra("INDEX", position)
        startActivity(intent)
    }

    override fun onResume() {
        super.onResume()
        val trainingController = TrainingController(context)
        val trainings: MutableList<TrainingStatistics> = trainingController.loadTrainings()
        val listDate: ArrayList<String> = ArrayList()
        GlobalScope.launch {
            trainings.forEach {
                listDate.add(it.date.toStringExtension(AppConstants.DATE_FORMAT))
            }
        }
        val adapter = MyArrayAdapter(requireContext(), android.R.layout.simple_list_item_1, listDate)
        binding.listView.adapter = adapter
    }

    private class MyArrayAdapter(context: Context, resource: Int, objects: ArrayList<String>): ArrayAdapter<String>(context, resource, objects) {
        override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
            val view = super.getView(position, convertView, parent)
            val textView = view.findViewById<View>(android.R.id.text1) as TextView
            textView.setTextColor(Color.WHITE)
            textView.setBackgroundColor(Color.BLACK)
            return view
        }
    }
}